### ISSUE TITLE
Multipliers in `hess` and `hess_coord` are keyword arguments, which default to `y0`.

### ISSUE THREAD

**Issue by abelsiqueira** [optimizers/CUTEst.jl#7]
I think that the multipliers in `hess` and `hess_coord` should not be keyword argument, because `hess(nlp, x, y)` seems more natural than `hess(nlp, x, y=y)`.

Also, I think that `hess(nlp, x)` defaulting to `hess(nlp, x, y0)` is error prone. We should default `y` to zero, or not use a default and create `hessObj` and `hessLagr` separately.

What do you think?

**Comment by dpo** [comment]
You're right. This could be a good use case of multiple dispatch:
```
hess(nlp, x)     # Hessian of objective; uses y=0
hess(nlp, x, y)  # Hessian of Lagrangian
```
and the same for `hess_coord()`.

**Comment by abelsiqueira** [comment]
Implemented in abelsiqueira@901d388.
